From 3f970049cb3a463df1065e273a577344539d7e36 Mon Sep 17 00:00:00 2001
From: Aditya Kumar Singh <quic_adisi@quicinc.com>
Date: Tue, 23 May 2023 17:49:29 +0530
Subject: [PATCH] mac80211: add Radio Measurement Action field values

Drivers which supports Tx power insertion will have to look
into the outgoing Radio Measurement packet. And based on the
type, driver will fill certain data in it. These values will
aid drivers in classifying the action code inside the
Radio Measurement action packet.

Hence, add Radio Measurement Action field values

Signed-off-by: Aditya Kumar Singh <quic_adisi@quicinc.com>
---
 include/linux/ieee80211.h | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/include/linux/ieee80211.h b/include/linux/ieee80211.h
index d7ef538b5094..e6e0547ba058 100644
--- a/include/linux/ieee80211.h
+++ b/include/linux/ieee80211.h
@@ -3530,6 +3530,16 @@ enum ieee80211_spectrum_mgmt_actioncode {
 	WLAN_ACTION_SPCT_CHL_SWITCH = 4,
 };
 
+/* RADIO_MEASUREMENT action code */
+enum ieee80211_radio_measurement_actioncode {
+	WLAN_ACTION_RADIO_MSR_RM_REQ = 0,
+	WLAN_ACTION_RADIO_MSR_RM_REP = 1,
+	WLAN_ACTION_RADIO_MSR_LINK_MSR_REQ = 2,
+	WLAN_ACTION_RADIO_MSR_LINK_MSR_REP = 3,
+	WLAN_ACTION_RADIO_MSR_NBOR_REP_REQ = 4,
+	WLAN_ACTION_RADIO_MSR_NBOR_REP_REP = 5,
+};
+
 /* HT action codes */
 enum ieee80211_ht_actioncode {
 	WLAN_HT_ACTION_NOTIFY_CHANWIDTH = 0,
-- 
2.17.1

